includes/class-linkbacks-mf2-handler.php

```diff
@@ -127,75 +130,213 @@ public static function generate_commentdata( $commentdata ) {
+		}
+
+		// Check and parse for location property
+		if ( array_key_exists( 'location', $properties ) ) {
+			$location = $properties['location'];
+			if ( is_array( $location ) ) {
+				if ( array_key_exists( 'latitude', $location ) ) {
+					$commentdata['comment_meta']['geo_latitude'] = $location['latitude'];
+				}
+				if ( array_key_exists( 'longitude', $location ) ) {
+					$commentdata['comment_meta']['geo_longitude'] = $location['longitude'];
+				}
+				if ( array_key_exists( 'name', $location ) ) {
+					$commentdata['comment_meta']['geo_address'] = $location['name'];
+				}
+			} else {
+				if ( substr( $location, 0, 4 ) == 'geo:' ) {
+					$geo = explode( ':', substr( urldecode( $location ), 4 ) );
+					$geo = explode( ';', $geo[0] );
+					$coords = explode( ',', $geo[0] );
+					$commentdata['comment_meta']['geo_latitude'] = trim( $coords[0] );
+					$commentdata['comment_meta']['geo_longitude'] = trim( $coords[1] );
+				} else {
+					$commentdata['comment_meta']['geo_address'] = $location;
+				}
+			}
+		}
+
 		// check rsvp property
-		if ( self::check_mf_attr( 'rsvp', $properties ) ) {
-			$commentdata['comment_meta']['semantic_linkbacks_type'] = wp_slash( 'rsvp:' . $properties['rsvp'][0] );
+		if ( isset( $properties['rsvp'] ) ) {
+			$commentdata['comment_meta']['semantic_linkbacks_type'] = wp_slash( 'rsvp:' . $properties['rsvp'] );
 		} else {
 			// get post type
 			$commentdata['comment_meta']['semantic_linkbacks_type'] = wp_slash( self::get_entry_type( $commentdata['target'], $entry, $mf_array ) );
 		}
+		$blacklist = array( 'name', 'content', 'summary', 'published', 'updated', 'type', 'url', 'comment', 'bridgy-omit-link' );
+		$blacklist = apply_filters( 'semantic_linkbacks_mf2_props_blacklist', $blacklist );
+		foreach ( $properties as $key => $value ) {
+			if ( ! in_array( $key, $blacklist ) ) {
+				$commentdata['comment_meta'][ 'mf2_' . $key ] = $value;
+			}
+		}
+		$commentdata['comment_meta'] = array_filter( $commentdata['comment_meta'] );
 
 		return $commentdata;
 	}
 
+	public static function convert_time( $time ) {
+		$time = strtotime( $time );
+		// If it can't read the time it will return null which will mean the comment time will be set to now.
+		if ( $time ) {
+			return get_date_from_gmt( date( 'Y-m-d H:i:s', $time ), 'Y-m-d H:i:s' );
+		}
+		return null;
+	}
+
+	public static function get_property( $key, $properties ) {
+		if ( isset( $properties[ $key ] ) && isset( $properties[ $key ][0] ) ) {
+			if ( is_array( $properties[ $key ] ) ) {
+					$properties[ $key ] = array_unique( $properties[ $key ] );
+			}
+			if ( 1 === count( $properties[ $key ] ) ) {
+						return $properties[ $key ][0];
+			}
+			return $properties[ $key ];
+		}
+		return null;
+	}
+
+	/**
+	 * Is string a URL.
+	 *
+	 * @param array $string
+	 * @return bool
+	*/
+	public static function is_url( $string ) {
+		if ( ! is_string( $string ) ) {
+			return false;
+		}
+		return preg_match( '/^https?:\/\/.+\..+$/', $string );
```

Comment:
can you add the possibility to test with local urls? perhaps it is ok to remove the check of the `dot`